proposals/0001-content-steering.md

```diff
@@ -0,0 +1,98 @@
+
+An example of how a typical interaction with the player, steering server and multiple CDNs is provided from the DASH IF spec.
+
+![Example DASH Content Steering Flow Diagram](./ContentSteering/DASH-Content-Steering-Flow.png)
+
+For more details on how this accomplished for each format, see the appropriate specification:
+* [HLS Content Steering Specifiation](https://developer.apple.com/streaming/HLSContentSteeringSpecification.pdf)
+* [DASH Content Steering Specification](https://dashif.org/docs/DASH-IF-CTS-00XX-Content-Steering-Community-Review.pdf)
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+To implement this feature in video.js, there will be 4 main components.
+* Parse `#EXT-X-CONTENT-STEERING` tags and associated properties in the m3u8-parser and expose them in the manifest object.
+* Parse `<ContentSteering>` tags and associated properties in the mpd-parser and expose them as well.
+* Add the ability to process the associated tags in VHS and request and refresh the content steering manifest.
+  - This would likely occur at the playlist-loader level as to more easily separate the DASH and HLS specific logic.
+* Process the content steering manifest and apply steering logic to both DASH and HLS content.
+  - This logic can be applied in common areas where applicable, but should be pushed up into the respective loaders when required.
+
+Both specifications (linked above) enumerate detailed steps on the required logic for handling both the steering manifest, new tags and associated properties. This logic must be implemented in a way specific to each format when necessary. For example, if required, [HLSv2 7.2 has specific requirements for pathway cloning](https://datatracker.ietf.org/doc/draft-pantos-hls-rfc8216bis/).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* This may not be a useful feature for *all* users of video.js.
+* The work required is substantial.
+* Could potentially cause regressions in core libraries such as the parsers or VHS. 
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+* This design is the most applicable to our current architecture, the details about which specific class to implement the may change as development progresses, but the main components will remain the same.
+* A separate content steering parser is an alternative that could keep some of the logic out of VHS for more clarity, but may not be necessary due to the small amount of parsing required.
+* Not developing this feature could put is behind some of the other open source players available.
+
+# Prior art
+[prior-art]: #prior-art
+
+* This feature is already implemented in DASH.js and hls.js, both can be a useful reference.
+* This feature is currently being discussed with the research team at Brightcove and has been the subject of papers being presented at video industry conferences such as IBC and demuxed.
+
+# Unresolved questions
```

Comment:
Another question I may have:

Will this be an optional feature in VJS? What about in the Birghtcove player? Will certain features, such as pathway cloning, be optional as well?

Reply:
I think this will be optional only in the sense that, if a manifest includes the content steering tags, we will engage the specified logic, similar to other tag specific behavior. Pathway cloning will be similar, however we still need to determine if that will be part of the initial implementation of the feature.